### PR TITLE
feat(sdk): add `Bucket.copy()` for `tf-azure` target

### LINKED ISSUE #4397: Implement `cloud.Bucket` inflight method `copy` for `tf-azure` target
```js
/**
 * Copy an object to a new location in the bucket. If the destination object
 * already exists, it will be overwritten.
 *
 * @throws Will throw if the `src` object doesn't exist.
 */
inflight copy(src: str, dst: str): void;
```

### PR DESCRIPTION
**Pull request by garysassano**
Closes #4397

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Comment by tsuf239** [comment]
will be reviewed after merging in the copy wing test in  https://github.com/winglang/wing/pull/4563

**Comment by tsuf239** [comment]
Can you please merge in main?

**Comment by garysassano** [comment]
Waiting for #4691 to get merged so I run the test in Azure.

**Comment by garysassano** [comment]
Waiting for #4691 to get merged to be able to test on Azure.

**Comment by garysassano** [comment]
Ready to ship.

azure-test-pass

**Comment by garysassano** [comment]
Ready to 🚢

azure-test-pass

File: libs/wingsdk/src/target-tf-azure/bucket.inflight.ts
Comment (tsuf239) [inline libs/wingsdk/src/target-tf-azure/bucket.inflight.ts:275]: what is the difference between `getBlobClient` to `getBlockBlobClient`?

Comment (garysassano) [reply libs/wingsdk/src/target-tf-azure/bucket.inflight.ts:275]: `BlobClient` is a generic client that works for all [blob types](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction#blobs), so it's generally used for read operations where you don't know the type of the blob you are accessing.

`BlockBlobClient` is a specialized client that works only for block blobs. This means that you must be sure that the blob on which you are operating is of the correct type. As you can imagine, type-specific clients have a wider set of methods for manipulating a particular blob type.

Comment (tsuf239) [reply libs/wingsdk/src/target-tf-azure/bucket.inflight.ts:275]: and why are we sure that the dest key is a block blob type?

Comment (garysassano) [reply libs/wingsdk/src/target-tf-azure/bucket.inflight.ts:275]: @tsuf239 Because you create the destination blob. You don't know what you are gonna read, but you definitely know what you want to write.

Comment (garysassano) [reply libs/wingsdk/src/target-tf-azure/bucket.inflight.ts:275]: @tsuf239 Because you are the one creating the destination blob. Basically, don't know what you are gonna read, but you definitely know what you want to write.

Comment (garysassano) [reply libs/wingsdk/src/target-tf-azure/bucket.inflight.ts:275]: @tsuf239 Because you are the one creating the destination blob. Basically, you don't know what you are gonna read, but you definitely know what you want to write.

File: libs/wingsdk/src/target-tf-azure/bucket.inflight.ts
Comment (eladb) [inline libs/wingsdk/src/target-tf-azure/bucket.inflight.ts:281]: Is that the only error. I would chain the original error message to the string:

```suggestion
        `Unable to copy. ${error.message} (srcKey=${srcKey}).`
```

**Review by eladb — APPROVED** [review]

**Pull request merged**
